pages/instructorFileEditor/instructorFileEditor.js

```diff
@@ -127,11 +158,11 @@ router.post('/', (req, res, next) => {
     if (!res.locals.authz_data.has_course_permission_own) return next(new Error('Insufficient permissions'));
 
     let fileEdit = {
-        editID: req.body.file_edit_id,
         userID: req.body.file_edit_user_id,
         courseID: req.body.file_edit_course_id,
         dirName: req.body.file_edit_dir_name,
         fileName: req.body.file_edit_file_name,
+        origHash: req.body.file_edit_orig_hash,
```

Comment:
Apologies for missing this the first time around, but this pattern of getting data from the client is generally risky (I didn’t check if it’s actually a security hole in this specific case). The problem is that, for example, `courseID` could have been changed by the client to point to a different course, and it’s very easy for us to fail to check for this. My rule of thumb is that:

* variables in `res.locals` have been through our `authz` functions and so can all be trusted.
* variables in `req.body` are basically just whatever random thing a potentially-evil client sent to us, and they should be treated with extreme skepticism 😄

In this case, I suggest using `res.locals.user.id` and `res.locals.course.id` rather than values from the client. I think that `dir_name`, `file_name`, etc are fine because you are already treating them as untrusted below (e.g., checking that `dir_name` is inside the course dir).

This type of security flaw is one that I’ve made many times — I find it easy to forget that the variable we put into the EJS is not necessarily the variable we will get back in the POST request.